calico_node/startup/startup.go

```diff
@@ -347,19 +370,25 @@ func parseIPEnvironment(envName, envValue string, version int) *net.IPNet {
 	}
 	message("Using IPv%d address from environment: %s=%s", ip.Version(), envName, envValue)
 
-	return ip
+	return ip.String()
 }
 
 // validateIP checks that the IP address is actually on one of the host
 // interfaces and warns if not.
-func validateIP(ipn *net.IPNet) {
+func validateIP(ipn string) {
 	// No validation required if no IP address is specified.
-	if ipn == nil {
+	if ipn == "" {
 		return
 	}
 
+	_, cidr, err := net.ParseCIDR(ipn)
+	if err != nil {
```

Comment:
I think it's valid for the IP to just be an IP - in which case we assume a /32 or /128 depending on the version.

So best to use the ParseCIDROrIP method in the net package.